src/checkpoint/container/vector_serialize.h

```diff
@@ -100,6 +100,22 @@ void serialize(Serializer& s, std::vector<bool, VectorAllocator>& vec) {
   }
 }
 
+template <
+  typename SerializerT,
+  typename T,
+  typename VectorAllocator,
+  typename = std::enable_if_t<
+    std::is_same< // is_footprintable
+      size_t,
+      decltype(checkpoint::getMemoryFootprint(std::declval<T>()))
+    >::value
+  >
+>
```

Comment:
OK, looking at the rest of this file's code, I think it should maybe first be applied to specializations of `dispatch::serializeArray` instead of `serialize(vector<T>)`. Also, I think the sense of the condition is reversed - we want to consider `sizeof(T)` when `!is_footprintable<T>`